### PR TITLE
FOGL-2592: Allow login using certificate

### PR DESCRIPTION
**Pull request by mshariq-nerd**
Login using certificate

### CODE REVIEW

File: src/app/components/layout/certificate-base-login/certificate-base-login.component.ts [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    const certificate_modal = <HTMLDivElement>document.getElementById('certificate-login-modal');
+    if (isOpen) {
+      certificate_modal.classList.add('is-active');
+      return;
+    }
+    certificate_modal.classList.remove('is-active');
+  }
+
+  selectOptionOfCertificateLogin() {
+    this.resetForm();
```
Comment (monika-Dianomic) [inline src/app/components/layout/certificate-base-login/certificate-base-login.component.ts]: `certificateContent` should also set to default in this case.

File: src/app/components/layout/certificate-base-login/certificate-base-login.component.html [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      </div>
+      <form [formGroup]="form" id="ngForm" (ngSubmit)="LoginWithCertificate()" novalidate>
+        <div class="columns">
+          <div class="column">
+            <div class="field is-horizontal">
+              <div class="field-label has-text-left">
+                <label *ngIf="showBrowseCertificate" class="label">Browse Certificate</label>
+                <label *ngIf="!showBrowseCertificate" class="label disabled">Browse Certificate</label>
+              </div>
+              <div class="field-body disabled">
```
Comment (monika-Dianomic) [inline src/app/components/layout/certificate-base-login/certificate-base-login.component.html]: No need of `disabled` class here.

File: src/app/components/layout/certificate-base-login/certificate-base-login.component.ts [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+          } else {
+            this.alertService.error(error.statusText);
+          }
+        });
+  }
+
+  protected resetForm() {
+    this.form.get('certificate').setValue('');
+    this.form.get('certificateText').setValue('');
+    this.isCertificateExt = true;
```
Comment (monika-Dianomic) [inline src/app/components/layout/certificate-base-login/certificate-base-login.component.ts]: `this.certificateContent` should also set to default here.

**Comment by billehunt** [comment]
Looks good to me. I suspect we'll need to change when customers actually use, but that's fine.

**Pull request merged**